docs/jobs.md

```diff
@@ -0,0 +1,432 @@
+day-of-week blank.
+```
+
+To schedule jobs given an interval:
+
+Examples:
+
+```
+@cron 0 0 0 1 1 * # Run once a year, midnight, Jan. 1st
+@cron 0 0 0 1 1 * # Run once a year, midnight, Jan. 1st
+@cron 0 0 0 1 * * # Run once a month, midnight, first of month
+@cron 0 0 0 * * 0 # Run once a week, midnight on Sunday
+@cron 0 0 0 * * * # Run once a day, midnight
+@cron 0 0 * * * * # Run once an hour, beginning of hour
+```
+
+
+### `@interval` syntax
+
+The `@interval` launcher uses the same syntax as golang's `time.ParseDuration`:
+
+```
+A duration string is a possibly signed sequence of decimal numbers, each with
+optional fraction and a unit suffix, such as "300ms", "1.5h" or "2h45m". Valid
+time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+```
+
+Examples
+
+```
+@interval 1.5h   # schedules every 1 and a half hours
+@interval 30m10s # schedules every 30 minutes and 10 seconds
+```
+
+
+### `@event` syntax
+
+The `@event` syntax is not determined yet. Its main purpose will be to describe job scheduling after a filesystem or database modification.
+
+
+### Launcher rate limitation
+
+Every launcher has a rate limitation policy to prevent launchers from spawning too many jobs at the same time. The specific rules are not yet decided, but they should work along these two properties:
+
+  - *parallel limitations*: each launcher will have a limited number of workers it is allowed to queue in parallel
+  - *back pressure*: each launcher should have a backpressure policy to drop job spawning when not necessary. For instance: 
+    - in the case of an `@event` launcher, a job doing an external API call should not be spawned if another one is already running for another close event.
+    - when no worker is availabe, or the queue has too many elements, it can decide to drop the job action (given some informations)
+
+
+Error Handling
+--------------
+
+Jbos can fail to execute their task. We have two ways to parametrize how they should behave in suche cases.
+
+### Retry
+
+A retry count can be optionnaly specified to ask the worker to re-exute the task if it has failed.
+
+Each retry is executed after a configurable delay. The try count is part of the attributes of the job. Also, each occuring error is kept in the `errors` field containing all the errors that may have happened.
+
+### Timeout
+
+A worker may never end. To prevent this, a configurable timeout value is specified with the job.
+
+If a job does not end after the specified amount of time, it will be aborted. A timeout is just like another error from the worker and can provoke a retry if specified.
+
+
+Job Data
+--------
+
+Example and description of the attributes of a `io.cozy.jobs`:
+
+```js
+{
+  "worker": "sendmail",    // worker type name
+  "worker_id": "123123",   // worker id, if any
+  "launcher": "@cron",     // "@cron", "@interval", "@event" or ""
+  "launcher_id": "1234",   // launcher id, if any
```

Comment:
This is more a "debuggability" / "inspection" feature. I think it may be useful to known how / why the job has been launched.